### PR TITLE
Remove Openssl dependency from fuel-client

### PR DESCRIPTION
**Pull request by ControlCplusControlV**
Disables surf's default features (which compiles using openssl-sys) and instead enables rustls as a replacement. From my local testing it seems to have removed the need for openssl from fuel-gql client. But as fuel-gql-client is a dependency of other crates it would need a version bump to fully remove the openssl dependency

Would it then be better to version bump fuel-gql client in this PR, or save that for a full release to fully remove openssl?

### CODE REVIEW

**Comment by ControlCplusControlV** [comment]
Done as part of #133

**Comment by Voxelot** [comment]
In our release process, we always bump the versions in a separate PR once we're ready to schedule the release. That way we can review all changes accumulated on master and verify the version bump follows semver conventions.

File: fuel-client/Cargo.toml [hunk 1]
```diff
@@ -28,7 +28,7 @@ hex = "0.4"
 itertools = "0.10"
 serde = { version = "1.0", features = ["derive"] }
 serde_json = { version = "1.0", features = ["raw_value"] }
-surf = "2.2"
+surf = {version = "2.2",default-features = false ,features = ["h1-client-rustls"] }
```
Comment (adlerjohn) [inline fuel-client/Cargo.toml:31]:
```suggestion
surf = { version = "2.2", default-features = false, features = ["h1-client-rustls"] }
```

**Review by adlerjohn — COMMENTED** [review]
The Cargo lockfile should also be committed, no? I imagine there will be changes, specifically it should no longer have `openssl-sys`.

**Review by Voxelot — APPROVED** [review]

**Comment by Voxelot** [comment]
[nit] we may be able to remove the openssl installation steps from the repo README.md / CONTRIBUTING.md

**Comment by ControlCplusControlV** [comment]
Was waiting on a version bump to confirm that before removing that, because openssl-sys is still a dependency of older (latest release) fuel-gql-client versions used by other crates.

**Comment by Voxelot** [comment]
Makes sense, we can defer that to another PR after this is released.

**Review by adlerjohn — CHANGES_REQUESTED** [review]
The lockfile doesn't actually seem to have `openssl-sys` removed?

**Review by adlerjohn — APPROVED** [review]
NVM, I see the comment on a circular dependency

**Comment by Voxelot** [comment]
This circular dep should get resolved when https://github.com/FuelLabs/fuels-rs/issues/62 is finished allowing the indexer to upgrade it's version of fuels-rs.

**Pull request merged**